clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -223,7 +223,7 @@ public synchronized void maybeAddPartitionToTransaction(TopicPartition topicPart
         if (isPartitionAdded(topicPartition) || isPartitionPendingAdd(topicPartition))
             return;
 
-        log.debug("{}Begin adding new partition {} to transaction", logPrefix, topicPartition);
+        log.trace("{}Begin adding new partition {} to transaction", logPrefix, topicPartition);
```

Comment:
This can probably be left at `debug`. Before it was very spammy because we didn't check if the partition was already inside `newPartitionsInTransaction`.